cranelift/jit/src/memory/system.rs

```diff
@@ -49,7 +53,51 @@ impl PtrLen {
         })
     }
 
-    #[cfg(all(not(target_os = "windows"), not(feature = "selinux-fix")))]
+    // macOS ARM64: Use mmap with MAP_JIT for W^X policy compliance.
+    #[cfg(all(
+        target_arch = "aarch64",
+        target_os = "macos",
+        not(feature = "selinux-fix")
+    ))]
+    fn with_size(size: usize) -> io::Result<Self> {
+        assert_ne!(size, 0);
+        let alloc_size = region::page::ceil(size as *const ()) as usize;
+
+        let ptr = unsafe {
+            libc::mmap(
+                ptr::null_mut(),
+                alloc_size,
+                libc::PROT_READ | libc::PROT_WRITE,
+                libc::MAP_PRIVATE | libc::MAP_ANON | libc::MAP_JIT,
```

Comment:
We should only pass MAP_JIT for pages that will actually end up being executable in the end.